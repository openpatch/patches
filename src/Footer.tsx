import { css } from "@emotion/react";
import { ReactNode, useEffect, useState } from "react";
import { Box } from "./Box";

export type FooterProps = {
  links?: {
    href: string;
    label: string;
  }[];
  copyright?: ReactNode;
};

export const Footer = ({ links, copyright = "OpenPatch" }: FooterProps) => {
  const [year, setYear] = useState<number | null>(null);
  useEffect(() => {
    setYear(new Date().getFullYear());
  }, []);
  return (
    <Box as="footer" backgroundColor="primary.900">
      <Box
        maxWidth="large"
        mx="auto"
        px={["small", "medium", "medium"]}
        py={["xsmall", "small"]}
      >
        {links && (
          <Box as="nav" aria-label="footer-menu">
            <ul
              css={(theme) => css`
                list-style: none;
                display: flex;
                align-items: center;
                padding: 0;
                margin-top: 0;
                margin-bottom: ${theme.space.xxsmall};
                flex-wrap: wrap;
                margin-left: -${theme.space.xsmall};
              `}
            >
              {links?.map(({ href, label }) => (
                <li
                  key={href}
                  css={css`
                    margin: 4px;
                  `}
                >
                  <Box
                    as="a"
                    fontWeight="semibold"
                    borderRadius="standard"
                    fontSize="small"
                    padding="xsmall"
                    css={(theme) => css`
                      text-decoration: none;

                      :hover {
                        color: ${theme.colors.primary["50"]};
                        background-color: ${theme.colors.primary["800"]};
                      }
                    `}
                    textColor="primary.200"
                    href={href}
                  >
                    {label}
                  </Box>
                </li>
              ))}
            </ul>
          </Box>
        )}
        {copyright && (
          <Box fontSize="small" textColor="primary.300" fontWeight="semibold">
            &copy; {year} {copyright}
          </Box>
        )}
      </Box>
    </Box>
  );
};
