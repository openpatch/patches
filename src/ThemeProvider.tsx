import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
