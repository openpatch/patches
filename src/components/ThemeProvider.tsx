import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
