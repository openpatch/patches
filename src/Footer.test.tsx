import React from "react";
import { render } from "@testing-library/react";
import { Footer, FooterProps } from "./Footer";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FooterProps) =>
  render(
    <ThemeProvider>
      <Footer {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
