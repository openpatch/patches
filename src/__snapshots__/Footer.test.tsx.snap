// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background-color: #014D40;
}

.emotion-1 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
}

@media screen and (min-width: 768px) {
  .emotion-1 {
    padding-left: 24px;
    padding-right: 24px;
    padding-top: 16px;
    padding-bottom: 16px;
  }
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  font-size: 14px;
  color: #65D6AD;
  font-weight: 600;
}

<div>
  <div
    id="patches-root"
  >
    <footer
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
          font-size="small"
          font-weight="semibold"
        >
          Â© 
          2022
           
          OpenPatch
        </div>
      </div>
    </footer>
  </div>
</div>
`;
